staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -120,38 +135,146 @@ func NewCloudNodeController(
 		recorder:                  recorder,
 		cloud:                     cloud,
 		nodeStatusUpdateFrequency: nodeStatusUpdateFrequency,
+		nodesLister:               nodeInformer.Lister(),
+		nodesSynced:               nodeInformer.Informer().HasSynced,
+		workqueue:                 workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Nodes"),
 	}
 
 	// Use shared informer to listen to add/update of nodes. Note that any nodes
 	// that exist before node controller starts will show up in the update method
 	cnc.nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    func(obj interface{}) { cnc.AddCloudNode(context.TODO(), obj) },
-		UpdateFunc: func(oldObj, newObj interface{}) { cnc.UpdateCloudNode(context.TODO(), oldObj, newObj) },
+		AddFunc:    cnc.enqueueNode,
+		UpdateFunc: func(oldObj, newObj interface{}) { cnc.enqueueNode(newObj) },
```

Comment:
@andrewsykim updated! 
I think probably we don't need `AddCloudNode` function anymore since it is basically equivalent to `UpdateCloudNode ` but with different logs 🤔